excercises/square_root/sqrt_error.go

```diff
@@ -0,0 +1,28 @@
+package main
+
+import (
+	"fmt"
+)
+
+type ErrNegativeSqrt float64
+
+func (e ErrNegativeSqrt) Error() string{
+	return "caanot Sqrt negative number:"
+}
+
+func Sqrt(x ErrNegativeSqrt) ErrNegativeSqrt {
+	var z,before ErrNegativeSqrt = 1.0,2.0
+	for ; before!=z;{
+		before = z
+		z -= (z*z - x) / (2*z)
+	}
+	return z
+
+}
+
+
+func main() {
+	var a,b ErrNegativeSqrt = 100.0,-4.0
+	fmt.Println(Sqrt(a))
```

Comment:
@lokanadham100 
For 100.0 I am getting an error although I shouldn't be getting error.
I don't understand why this is happening.
What am I doing wrongly

Reply:
https://github.com/davidhoo/go-tour/blob/master/exercise-errors.go